bobcat/bb-aem-touch-ui/src/main/java/com/cognifide/qa/bb/aem/touch/siteadmin/aem61/SiteadminToolbar.java

```diff
@@ -0,0 +1,191 @@
+
+  @Inject
+  private WebDriver driver;
+
+  @Inject
+  private WebElementUtils webElementUtils;
+
+  @Inject
+  private BobcatWait wait;
+
+  @FindBy(css = "button.cq-siteadmin-admin-actions-edit-activator[title='Open']")
+  private WebElement openButton;
+
+  @FindBy(css = "form.cq-siteadmin-admin-createpage")
+  private CreatePageActions createPageActions;
+
+  @FindBy(css = "button.coral-Button--graniteActionBar.cq-wcm-paste-activator")
+  private WebElement pastePageBtn;
+
+  @FindBy(css = "i.coral-Icon.coral-Icon--delete")
+  @Global
+  private WebElement deleteBtn;
+
+  @Global
+  @FindBy(css = "form.activation-wizard")
+  private CoralWizard activationWizard;
+
+  @Global
+  @FindBy(css = "form.activation-wizard")
+  private ReplicateLaterWindow replicateLaterWindow;
+
+  @Global
+  @FindBy(css = "form.move-page-wizard")
+  private MovePageWizard movePageWizard;
+
+  @Global
+  @FindBy(css = "div.coral-Modal.coral-Modal--notice")
+  private WebElement confirmationModal;
+
+  @Global
+  @FindBy(css = "div.foundation-ui-notification.coral-Alert.coral-Alert--info")
+  private WebElement successInfo;
+
+  public void openPage() {
+    openButton.click();
+  }
+
+  public void publishPage() {
+    getMoreBtn().click();
+    getPublishBtn().click();
+    clickButtonFromMoreSection(PUBLISH_LBL);
+    if (!successInfoAppeared()) {
+      activationWizard.submit();
+    }
+  }
+
+  private boolean successInfoAppeared() {
+    return driver.findElements(
+        By.cssSelector("div.foundation-ui-notification.coral-Alert.coral-Alert--info")).
+        size() > 0;
+  }
+
+  private WebElement getMoreBtn() {
+    return driver.findElement(By.cssSelector("i.coral-Icon.coral-Icon--more"));
+  }
+
+  private WebElement getPublishBtn() {
+    return driver.findElement(By.
+        cssSelector("a.cq-siteadmin-admin-actions-publish-activator[title='Publish']"));
+  }
+
+  private WebElement getDeleteBtn() {
+    if (!deleteBtn.isDisplayed()) {
+      getMoreBtn().click();
+      wait.withTimeout(1);
+    }
+    return deleteBtn;
+  }
+
+  private WebElement getUnpublishBtn() {
+    return driver.findElement(By.
+        cssSelector("a.cq-siteadmin-admin-actions-unpublish-activator[title='Unpublish']"));
+  }
+
+  private void clickButtonFromMoreSection(String buttonLabel) {
+    List<WebElement> elements = driver.findElements(By.cssSelector(BTN_LIST_SELECTOR));
+    elements.stream()
+        .filter(t -> t.getText().equals(buttonLabel))
+        .findFirst().get().click();
+  }
+
+  private void clickButtonInTheScope(WebElement scope, String buttonLabel) {
+    scope.findElements(By.tagName("button")).stream()
+        .filter(t -> buttonLabel.equals(t.getText()))
+        .findFirst().get().click();
+  }
+
+  public void unpublishPage() {
+    getMoreBtn().click();
+    getUnpublishBtn().click();
+    clickButtonFromMoreSection(UNPUBLISH_LBL);
+    if (!successInfo.isDisplayed() && confirmationModal.isDisplayed()) {
```

Comment:
this will be very long execution if successInfo and confirmationModal will be not present